designs/2019-typescript-eslintrc-support/README.md

```diff
@@ -0,0 +1,331 @@
+- Start Date: 2019-11-18
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Gareth Jones ([@g-rath](https://github.com/g-rath))
+
+# First-Class Support For `.eslintrc.ts`
+
+## Summary
+
+Support the ability to consume `.eslintrc.ts` configuration files natively.
+
+## Motivation
+
+Currently configuration settings are validated by JSON Schema, which while
+powerful does not have ubiquitous & high quality support across editors &
+tooling.
+
+It also tends to generate somewhat cryptic messages, especially when using
+complex elements such as combining `oneOf`, `anyOf`, and other such schemas.
+
+Meanwhile, TypeScript is a powerful & popular superset of Javascript that has
+high quality support as a first class language in the majority of editors, and
+is quickly gaining ubiquitous support in general.
+
+By allowing first-class support for `.eslintrc.ts`, the amount of configuration
+problems & general confusion over such experienced by users can be reduced,
+since it would enable us to provide tooling to type configuration files and so
+allow TypeScript to catch ESLint configuration errors at compile time, rather
+than run time.
+
+While we can already hack ways to use `.eslintrc.ts`, none of them lend
+themselves to ease-of-use in the same manner that ESLint encourages, and so
+first-class support for `.eslintrc.ts` will go long way towards encouraging
+community driven typing of ESLint plugins.
+
+Finally, ESLint is being adopted as the linter for TypeScript, meaning
+first-class support for `.eslintrc.ts` will help enable a positive feedback loop
+to help improve the two tools.
+
+There are some potential UX benefits for users that use & understand TypeScript,
+in the form
+
+## Detailed Design
+
+- Adds support for `.eslintrc.ts`
+- Requires uses to install `ts-node` manually adjacent to ESLint in
+  `node_modules`.
```

Comment:
My apologies my wording here is incorrect - it should say something to the effect of "It will load `ts-node` based on the standard [node resolution logic](https://nodejs.org/api/modules.html#modules_loading_from_node_modules_folders).

I have no opinion on the loading behaviour, other than that the current implementation I have is using plain `require` for simplicity.

I believe that puts it in in line w/ the behaviour you're described.

Reply:
I think that it's better if it uses a local-installed `ts-node` always, as aligning to the other pluggable stuff.